int64 timeout_ms
---
bool ready